app/controllers/admin/talentlms_controller.rb

```diff
@@ -9,18 +9,32 @@ class TalentlmsController < ApplicationController
     before_action :require_can_manage_config!
 
     def index
-      @config = Talentlms::Config.first_or_initialize
+      @config = config_scope.first_or_initialize
+      redirect_to new_admin_talentlm_path if @config.new_record?
+      redirect_to edit_admin_talentlm_path(@config) unless @config.new_record?
+    end
+
+    def new
+      @config = config_scope.first_or_initialize
```

Comment:
I think the use of `first_or_initialize` in `new` and then `create` in `create`, could be confusing... If `new` gets triggered when there is already a config, it will add another unreachable config, and then appear to lose the configuration as the first config is displayed...